service/task_manager_module.cc

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (C) 2024-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include "locator/tablets.hh"
+#include "replica/database.hh"
+#include "service/storage_service.hh"
+#include "service/task_manager_module.hh"
+#include "tasks/task_handler.hh"
+#include "tasks/virtual_task_hint.hh"
+
+namespace service {
+
+tasks::task_manager::task_group tablet_virtual_task::get_group() const noexcept {
+    return tasks::task_manager::task_group::tablets_group;
+}
+
+future<std::optional<tasks::virtual_task_hint>> tablet_virtual_task::contains(tasks::task_id task_id) const {
+    auto tables = get_table_ids();
+    for (auto table : tables) {
+        auto& tmap = _ss.get_token_metadata().tablets().get_tablet_map(table);
+        for (const locator::tablet_info& info : tmap.tablets()) {
+            if (info.repair_task_info.is_valid() && info.repair_task_info.tablet_task_id.uuid() == task_id.uuid()) {
```

Comment:
We need to yield, otherwise we might stall here, e.g., use coroutine::maybe_yield().